dspace-server-webapp/src/main/java/org/dspace/app/rest/ContentReportRestController.java

```diff
@@ -0,0 +1,226 @@
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.rest.webmvc.ControllerUtils;
+import org.springframework.hateoas.Link;
+import org.springframework.hateoas.RepresentationModel;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * This controller receives and dispatches requests related to the
+ * contents reports ported from DSpace 6.x (Filtered Collections
+ * and Filtered Items).
+ * @author Jean-François Morin (Université Laval)
+ */
+@RestController
+@RequestMapping("/api/" + RestModel.CONTENT_REPORT)
+public class ContentReportRestController implements InitializingBean {
+
+    private static final Logger log = org.apache.logging.log4j.LogManager.getLogger();
+
+    @Autowired
+    private DiscoverableEndpointsService discoverableEndpointsService;
+    @Autowired
+    private ConverterService converter;
+    @Autowired
+    private ContentReportRestRepository contentReportRestRepository;
+    @Autowired
+    private ContentReportService contentReportService;
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        discoverableEndpointsService
+            .register(this, List.of(Link.of("/api/" + RestModel.CONTENT_REPORT, RestModel.CONTENT_REPORT)));
+    }
+
+    @RequestMapping(method = RequestMethod.GET)
+    public ContentReportSupportResource getContentReportSupport() {
+        ContentReportSupportRest contentReportSupportRest = contentReportRestRepository.getContentReportSupport();
+        return converter.toResource(contentReportSupportRest);
+    }
+
+    /**
+     * GET-based endpoint for the Filtered Collections contents report.
+     * This method also serves as a feed for the HAL Browser infrastructure.
+     * @param filters querying filters received as a comma-separated string
+     * or as a multivalued parameter
+     * @param request HTTP request
+     * @param response HTTP response
+     * @return the list of collections with their respective statistics
+     */
+    @PreAuthorize("hasAuthority('ADMIN')")
+    @GetMapping("/filteredcollections")
+    public ResponseEntity<RepresentationModel<?>> getFilteredCollections(
+            @RequestParam(name = "filters", required = false) List<String> filters,
+            HttpServletRequest request, HttpServletResponse response) throws IOException {
+        if (contentReportService.getEnabled()) {
```

Comment:
@jeffmorin : Just realized you are managing "disabling" this controller in a manual fashion.   I think you could replace all your 404 logic & this `if` statement with simply adding this:
```
@ConditionalOnProperty("contentreport.enable")
```
This can be added at the *TOP* of the Controller and it will disable the entire controller & all endpoints if the setting is `false`.  In other words, when the setting is false, Spring will default to having all endpoints in this controller respond with a 404.

As an example, see this: https://github.com/DSpace/DSpace/blob/main/dspace-iiif/src/main/java/org/dspace/app/iiif/IIIFController.java#L32-L33

Reply:
Done. This leads indeed to an interesting code simplification.

Reply:
I finally rolled back to manual configuration parameter management for two reasons:

(1) Using `@ConditionalOnProperty` foils my integration tests in `ContentReportRestRepositoryIT` by making everything in the controller return an error 404... even under correct/authorized conditions.
(2) With my manual parameter management, I get a consistent behaviour depending on the context (turned on = 200, unauthorized = 401, turned off = 404).

Reply:
@jeffmorin : Aha, I see.  I suspect that you just needed to enable the content reports in the Test Environment's local.cfg here...similar again to how IIIF works: https://github.com/DSpace/DSpace/blob/main/dspace-api/src/test/data/dspaceFolder/config/local.cfg#L94-L98

I forgot to mention, the only side-effect to `@ConditionalOnProperty` is that changing the value of the setting requires a *reboot*. This means that if the Test environment initially loads with the value set to "false" then there's no way to get the Controller to respond.  The way around that is to get the Test Environment to set the value to "true" globally in that `local.cfg` above.

If you'd rather manage it manually, I'm not against it.  I just saw that this would be a simplification of your code (as it sounds like you saw as well)

Reply:
I didn't change my mind about simplifying my code, don't worry... I will give this addition to that local.cfg file a try right away.

Reply:
What's that??? when the parameter is set to true in local.cfg, the fonctionality becomes available alright, but it's now impossible to disable it and receive proper error messages accordingly: my tests always receive a 200 response!

Reply:
@jeffmorin : Ok, maybe I thought this would be a simple change and it's *not*.  I'm fine with you reverting it for now.  I'm not sure *why* this isn't working (as it works fine for IIIF in the code that I linked)... but, I don't have time to dig deeper immediately (as I have several feature PRs I'm helping finalize at once)

Reply:
OK. I read a bit about this annotation, and all it handles is whether or not a Spring should be initialized, depending on a given parameter value. After initializing the bean if the parameter value is true, there is still need for manual control, otherwise everything will pass if the parameter value is not further checked prior to performing some operation in the annotated bean (here the controller).

I understand that I could keep the annotation, and this would result in errors 404 everywhere (in the controller's scope, of course) if the parameter value is false. This is OK. But if the parameter value is true, then I need a behaviour that adjusts to the current parameter value. This is provided by my manual parameter checking, not the annotation. Maybe this is not needed in the case of `IIIFController`, which would explain why the annotation is a good fit there.

I will roll back (again) to my manual parameter checking, but with a better knowledge of why things go wrong when I use `@ConditionalOnParameter` alone.